### PR TITLE
Added optuna for dqn

### PR DESCRIPTION
**Pull request by Thematiq**

### CODE REVIEW

File: optuna_dqn.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        'board_size': BOARD_SIZE,
+        'seed': RANDOM_STATE,
+        'gamma': trial.suggest_float(0, 1),
+        'learning_rate': trial.suggest_float(1e-5, 1e-2),
+        'epsilon': trial.suggest_float(0, 1),
+        'epsilon_decay': trial.suggest_float(0, 1),
+        'soft_update': trial.suggest_float(0, 1),
+        'capacity': trial.suggest_int(1e2, 5e4),
+        'experience_replay_steps': trial.suggest_int(1, 10)
+    }
```
Comment (m-wojnar) [inline optuna_dqn.py:84]: I suggest values:
- `gamma` in 0.5 - 1.0,
- `epsilon` close to 1 (e.g., 0.9 - 1.0),
- `epsilon_decay` very close to 1 (e.g., 0.95 - 0.9999),
- `soft_update` very close to 0 (e.g., 0.0001 - 0.05).

Maybe it is also worth to have random `seed`?

File: optuna_dqn.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+
+def objective(trial):
+    params = sample_dqn_params(trial)
+    random_rounds = trial.suggest_int(5, 100)
+    minmax_rounds = trial.suggest_int(5, 50)
```
Comment (m-wojnar) [inline optuna_dqn.py]: I think that the number of rounds must mu much higher, like 1e3 to 1e5
